homeassistant/components/motionblinds_ble/config_flow.py

```diff
@@ -0,0 +1,215 @@
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import BluetoothServiceInfoBleak
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import (
+    CONF_BLIND_TYPE,
+    CONF_LOCAL_NAME,
+    CONF_MAC_CODE,
+    DOMAIN,
+    ERROR_COULD_NOT_FIND_MOTOR,
+    ERROR_INVALID_MAC_CODE,
+    ERROR_NO_BLUETOOTH_ADAPTER,
+    ERROR_NO_DEVICES_FOUND,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({vol.Required(CONF_MAC_CODE): str})
+
+
+class FlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Motionblinds BLE."""
+
+    def __init__(self) -> None:
+        """Initialize a ConfigFlow."""
+        self._discovery_info: BluetoothServiceInfoBleak | BLEDevice | None = None
+        self._mac_code: str | None = None
+        self._display_name: str | None = None
+        self._blind_type: MotionBlindType | None = None
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> ConfigFlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug(
+            "Discovered Motionblinds bluetooth device: %s", discovery_info.as_dict()
+        )
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+
+        self._discovery_info = discovery_info
+        self._mac_code = get_mac_from_local_name(discovery_info.name)
+        self._display_name = DISPLAY_NAME.format(mac_code=self._mac_code)
+        self.context["local_name"] = discovery_info.name
+        self.context["title_placeholders"] = {"name": self._display_name}
+
+        return await self.async_step_confirm()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            mac_code = user_input[CONF_MAC_CODE]
+            # Discover with BLE
+            try:
+                await self.async_discover_motionblind(mac_code)
+            except NoBluetoothAdapter:
+                return self.async_abort(reason=EXCEPTION_MAP[NoBluetoothAdapter])
+            except NoDevicesFound:
+                return self.async_abort(reason=EXCEPTION_MAP[NoDevicesFound])
+            except tuple(EXCEPTION_MAP.keys()) as e:
+                errors = {
+                    "base": (
+                        EXCEPTION_MAP[type(e)]
+                        if type(e) in EXCEPTION_MAP
+                        else str(type(e))
+                    )
+                }
+                return self.async_show_form(
```

Comment:
I'm missing these abort reasons and errors in the strings.json

Reply:
no_bluetooth_adapter and no_devices_found abort are in the strings.json? could_not_find_motor and invalid_mac_code are errors

```
    "abort": {
      "no_bluetooth_adapter": "No bluetooth adapter found",
      "no_devices_found": "Could not find any bluetooth devices"
    },
    "error": {
      "could_not_find_motor": "Could not find a motor with that MAC code",
      "invalid_mac_code": "Invalid MAC code"
    },
```